src/dstack/_internal/cli/utils/rich.py

```diff
@@ -122,3 +126,31 @@ def prepend_path(self, message, record):
                 path = path + "[/link]"
         message = f"[log.path]{path}[/] " + message
         return message
+
+
+class MultiItemStatus:
+    """An alternative to rich.status.Status that allows extra renderables below the spinner"""
+
+    def __init__(self, status: "RenderableType", *, console: Optional["Console"] = None) -> None:
+        self._spinner = Spinner("dots", text=status, style="status.spinner")
+        self._live = Live(
+            renderable=self._spinner,
+            console=console,
+            refresh_per_second=12.5,
+            transient=True,
+        )
+
+    def update(self, *renderables: "RenderableType") -> None:
+        self._live.update(renderable=Group(self._spinner, *renderables))
+
+    def __enter__(self) -> "MultiItemStatus":
+        self._live.start()
+        return self
+
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
```

Comment:
BTW, since we dropped 3.8 support, we can use stdlib types (`type` instead of `Type`) in new code